Kconfig

```diff
@@ -2005,6 +2005,90 @@ config FRAME_POINTER
 
 endmenu # Debug Options
 
+menuconfig  TRACE
+	bool "Enable tracepoints"
+	default n
+	depends on SCHED_INSTRUMENTATION_DUMP && DRIVERS_NOTE
+	---help---
+		If this option is enabled, tracepoints at critical code in
+		the kernel will be enabled.
+
+if TRACE
+
+config TRACE_BINFMT
+	bool "Enable tracepoints in binfmt"
+	default n
+
+config TRACE_BOARDS
+	bool "Enable tracepoints in boards"
+	default n
+
+config TRACE_CRYPTO
+	bool "Enable tracepoints in crypto"
+	default n
+
+config TRACE_DRIVERS
+	bool "Enable tracepoints in drivers"
+	default n
+
+config TRACE_INPUT
+	bool "Enable tracepoints in input"
```

Comment:
Why need to classify traces as modules? I think inside the module they can be configured dynamically

Reply:
In addition, it is best that this feature can be dynamically configured in the runtime.

Reply:
Done
I added tags to sched_note_string/printf to support runtime filtering, but it is still configurable at compile time